docs/protocols.ipynb

```diff
@@ -181,16 +145,112 @@
+    "|`cirq.apply_channel`| simulation |\n",
+    "|`cirq.has_channel`| simulation |\n",
+    "|`cirq.apply_mixture`| simulation |\n",
+    "|`cirq.mixture`|simulation  |\n",
+    "|`cirq.has_mixture`| simulation |\n",
+    "|`cirq.unitary`| simulation |\n",
+    "|`cirq.apply_unitary`| simulation |\n",
+    "|`cirq.has_unitary`| simulation |\n",
+    "|`cirq.has_stabilizer_effect`| simulation |\n",
+    "|`cirq.trace_distance_bound`| compilation |\n",
+    "|`cirq.equal_up_to_global_phase`| debugging, simulation, compilation |\n",
+    "|`cirq.approx_eq`| debugging, simulation, compilation |\n",
+    "|`cirq.measurement_key`| simulation, analysis   |\n",
+    "|`cirq.qid_shape`| circuit-construction, debugging, simulation |\n",
+    "|`cirq.resolve_parameters`| simulation, compilation, debugging |\n",
+    "|`cirq.parameter_names`| simulation, compilation, debugging |\n",
+    "|`cirq.parameter_symbols`| simulation, compilation, debugging |\n",
+    "|`cirq.commutes`| debugging, compilation, simulation |\n",
+    "|`cirq.decompose`|  debugging, compilation, simulation |\n",
+    "|`cirq.pauli_expansion`|  debugging, compilation, simulation |\n",
+    "|`cirq.inverse`| circuit-construction  |\n",
+    "|`cirq.phase`|  circuit-construction|\n",
+    "|`cirq.pow`| circuit-construction  |\n",
+    "|`cirq.mul`| circuit-construction |\n",
+    "|`cirq.circuit_diagram_info`| visualization |\n",
+    "|`cirq.qasm`| serialization |\n",
+    "|`cirq.quil`| serialization  |\n",
+    "|`cirq.json_serialization`| serialization |\n",
+    "\n",
+    "\n",
+    "### Quantum operator representation protocols\n",
+    "\n",
+    "The following family of protocols is an important and frequently used set of features of Cirq and it is worthwhile mentioning them and and how they interact with each other. They are, in the order of generality (starting with the least generic):\n",
+    "\n",
+    "* `*unitary`\n",
+    "* `*mixture`\n",
+    "* `*channel`\n",
+    "\n",
+    "All these protocols make it easier to work with different representations of quantum operators: finding that representation (`unitary`, `mixture`, `channel`), determining whether the operator has that representation (`has_*`) and applying them (`apply_*`) on a state vector. \n",
+    "\n",
+    "The `unitary` representation is the most special case, only applies when the quantum operator is unitary, and in this case represents it via the unitary matrix in the computational basis. We saw an example of the unitary protocol above.\n",
+    "\n",
+    "The `mixture` representation is a probabilistic representation of a quantum operator, where each possible unitary effect $U_i$ occurs with a certain probability $p_i$, where $\\sum p_i = 1$ In case the operator does not implement `SupportsMixture`, but does implement `SupportsUnitary`, `*mixture` protocols fall back to the `*unitary` methods, based on the simple correspondence: the $U$ is a mixture of a single unitary with probability $p=1$. \n"
```

Comment:
It might be worth explaining what each of these return (numpy array for unitary, for instance), since we are going into these methods in some depth.